### PR TITLE
Handle additional properties

### PR DESCRIPTION
**Pull request by drewbanin**
This PR makes `tap-codat` more resilient to undocumented and newly-added fields in the Codat API.

When a new field appears that is not present in the JSON Schema, this tap will log a human-readable INFO log.

### CODE REVIEW

File: tap_codat/streams.py [hunk 1]
```diff
@@ -26,6 +26,18 @@ def write_records(self, records):
         singer.write_records(self.tap_stream_id, records)
         self.metrics(records)
 
+    def log_additional_properties(self, ctx, records):
```
Comment (KAllan357) [inline tap_codat/streams.py]: Am I correct in my understanding that this is using the singer transform code to output an error so we can just log it out for visibility? If so, can we add a comment here to make that more clear?

File: tap_codat/streams.py [hunk 2]
```diff
@@ -26,6 +26,18 @@ def write_records(self, records):
...
+        schema = ctx.catalog.get_stream(self.tap_stream_id).schema.to_dict()
+        logged_error = False
+        for i, record in enumerate(records):
```
Comment (KAllan357) [inline tap_codat/streams.py]: I don't see `i` being used here. Do we need to use `enumerate`?

**Comment by drewbanin** [comment]
good points @KAllan357 -- thanks. Updated!

**Comment by KAllan357** [comment]
Thanks, this looks good.

**Pull request merged**